node/src/components/network.rs

```diff
@@ -599,24 +633,66 @@ where
                     // connection after a peer has closed the corresponding incoming connection.
                 }
 
+                // If given a key, determine validator status.
+                let validator_status = peer_consensus_public_key.as_ref().map(|public_key| {
+                    let status = self
+                        .validator_matrix
+                        .is_active_or_upcoming_validator(public_key);
+
+                    // Find the shared `Arc` that holds the validator status for this specific key.
+                    match self.incoming_validator_status.entry((**public_key).clone()) {
+                        // TODO: Use `Arc` for public key-key.
```

Comment:
After we update them, they'll be 32 or 33 bytes.

Reply:
That sounds much more reasonable - and it would have also affected other corner cases I had to fix where event sizes suddenly grew, often due to the size of `PublicKey`.

I'd leave this comment in still, whoever makes the public key change will then be able to clean it up?